ui/src/menu.tsx

```diff
@@ -0,0 +1,45 @@
+import {
+  default as BaseMenu,
+  MenuProps as BaseMenuProps,
+} from "@mui/material/Menu";
+import {
+  createElement,
+  MouseEvent,
+  ReactElement,
+  ReactNode,
+  useCallback,
+  useState,
+} from "react";
+
+interface MenuProps
+  extends Omit<BaseMenuProps, "anchorEl" | "open" | "onClick" | "onClose"> {
+  items: ReactNode;
+}
+
+export function useMenu(
```

Comment:
Interesting, are you basing this on an existing pattern or style guide from somewhere? I've not seen this "hook returning a component" pattern before, but from searching around I do see some mentions of it online.